importer/api/__init__.py

```diff
@@ -0,0 +1,95 @@
+"""
+Import OLX data into LORE.
+"""
+
+from glob import glob
+from datetime import datetime
+from shutil import rmtree
+from tempfile import mkdtemp
+from os.path import join
+import tarfile
+import zipfile
+
+from xbundle import XBundle, DESCRIPTOR_TAGS
+from lxml import etree
+
+from learningobjects.api import create_course, create_lox
+
+
+def import_course_from_file(filename, user_id):
+    """
+    Import OLX from .zip or tar.gz.
+
+    Args:
+        filename (unicode): Path to archive file (zip or .tar.gz)
+    Raises:
+        ValueError: Unable to find single path inside archive file.
+    Returns: None
+    """
+    tempdir = mkdtemp()
+    if filename.endswith(".tar.gz"):
+        course = tarfile.open(filename, "r")
+    elif filename.endswith(".zip"):
+        course = zipfile.ZipFile(filename, "r")
+    else:
+        raise ValueError("Unexpected file type (want tarball or zip).")
+    course.extractall(tempdir)
+    dirs = glob(join(tempdir, "*"))
+    if len(dirs) != 1:
+        raise ValueError("Unable to get course directory.")
+    import_course_from_path(dirs[0], user_id)
+    rmtree(tempdir)
```

Comment:
You nuke the tempdir, but this should also nuke the original file as well so we don't end up with a cluttered system.  Especially since we should be able to reconstruct the tar ball from the DB entries we are making.